6238_prototype/src/main/java/frc/robot/commands/Intake/RetractIntakeCommand.java

```diff
@@ -0,0 +1,31 @@
+package frc.robot.commands.Intake;
+
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.subsystems.BallSubsystem;
+
+public class RetractIntakeCommand extends CommandBase{
+    private final long timeLimit;
+    BallSubsystem ball;
+    long startTime;
+    public RetractIntakeCommand(BallSubsystem ball) {
+        this.ball = ball;
+        addRequirements(ball);
+        timeLimit = 2000;
+        startTime = System.currentTimeMillis();
+    }
+
+    @Override
+    public void execute() {
+        ball.retract();
+        double bottomMotorSpeed = 0.8;
+        double upperMotorSpeed = -0.1;
+        ball.setSpeed(bottomMotorSpeed, upperMotorSpeed);
+    }
+
+    @Override
+    public boolean isFinished() {
+        long currentTime = System.currentTimeMillis();
+        long timeElapsed = currentTime - startTime;
+        return timeElapsed > timeLimit;
+    }
```

Comment:
We need to call `setSpeed(0,0)` to ensure the motor does not continue to run.  I think you should just add an end method and call it there.